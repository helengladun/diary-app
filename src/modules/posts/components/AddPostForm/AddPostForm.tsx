import React, {ChangeEvent, FormEvent} from 'react';

interface IProps {
    submitHandler: (e: FormEvent<HTMLFormElement>) => void,
    changeHandler: (e: ChangeEvent<HTMLInputElement>) => void
    value: string
}

const AddPostForm = (props: IProps) => (
    <form className="main-form main-form_post" onSubmit={props.submitHandler}>
        <input className="primary-input" type="text" placeholder="Type name here..." onChange={props.changeHandler} value={props.value}/>
        <button className="btn-ordinary">
            <span className="btn-ordinary__text">Add new</span>
        </button>
    </form>
);

export default AddPostForm;