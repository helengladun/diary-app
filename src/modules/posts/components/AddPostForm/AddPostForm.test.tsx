import React from 'react';
import {shallow} from 'enzyme';
import AddPostForm from "./AddPostForm";

describe('AddCommentForm component', () => {
    const submitHandler = jest.fn();
    const changeHandler = jest.fn();
    const wrapper = shallow(<AddPostForm
        value={'text'}
        submitHandler={submitHandler}
        changeHandler={changeHandler}
    />);

    test('should render AddPostForm correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });
});

