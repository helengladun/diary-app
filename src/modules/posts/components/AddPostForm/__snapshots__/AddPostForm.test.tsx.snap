// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCommentForm component should render AddPostForm correctly 1`] = `ShallowWrapper {}`;
